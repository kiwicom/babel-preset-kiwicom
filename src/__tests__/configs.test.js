// @flow

import path from 'path';

import preset from '../index';

const supportedTargets = ['js', 'js-esm', 'flow'];

const apiMock = {
  assertVersion: version => {
    if (version !== 7) {
      throw new Error(
        'Only Babel API version 7 supported, given version: ' + version,
      );
    }
  },
};

expect.addSnapshotSerializer({
  test: () => true,
  print: value => {
    const cwd = path.join(__dirname, '..', '..'); // root of our Babel Preset
    return JSON.stringify(value, null, 2).replace(
      new RegExp(cwd, 'g'),
      '<BABEL_PRESET_ROOT>',
    );
  },
});

test.each(supportedTargets)(
  "preset with target '%s' emits correct config",
  target => {
    expect(preset(apiMock, { target })).toMatchSnapshot();
  },
);
