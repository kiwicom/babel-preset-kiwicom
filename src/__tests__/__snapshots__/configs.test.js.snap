// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`preset with target 'flow' emits correct config 1`] = `
{
  "presets": [],
  "plugins": [
    "<BABEL_PRESET_ROOT>/src/dev-declaration"
  ],
  "parserOpts": {
    "plugins": [
      "jsx",
      "flow",
      "flowComments",
      "dynamicImport",
      "classProperties",
      "classPrivateProperties",
      "nullishCoalescingOperator",
      "objectRestSpread",
      "optionalChaining",
      "numericSeparator"
    ]
  },
  "retainLines": true
}
`;

exports[`preset with target 'js' emits correct config 1`] = `
{
  "presets": [
    [
      "@babel/preset-env",
      {
        "modules": "commonjs",
        "targets": {
          "node": "current",
          "browsers": [
            "last 2 versions",
            "ie >= 11"
          ]
        }
      }
    ],
    "@babel/preset-react"
  ],
  "plugins": [
    "<BABEL_PRESET_ROOT>/src/dev-expression.js",
    "<BABEL_PRESET_ROOT>/src/kiwicom-js-invariant.js",
    "<BABEL_PRESET_ROOT>/src/kiwicom-js-warning.js",
    "@babel/plugin-transform-flow-strip-types",
    "@babel/plugin-proposal-class-properties",
    "@babel/plugin-proposal-nullish-coalescing-operator",
    "@babel/plugin-proposal-object-rest-spread",
    "@babel/plugin-proposal-optional-chaining",
    "@babel/plugin-proposal-numeric-separator",
    [
      "@babel/plugin-transform-runtime",
      {
        "useESModules": false
      }
    ],
    "@kiwicom/babel-plugin-orbit-components"
  ],
  "parserOpts": {
    "plugins": [
      "jsx",
      "flow",
      "flowComments",
      "dynamicImport"
    ]
  },
  "retainLines": false
}
`;

exports[`preset with target 'js-esm' emits correct config 1`] = `
{
  "presets": [
    [
      "@babel/preset-env",
      {
        "modules": false,
        "targets": {
          "node": "current",
          "browsers": [
            "last 2 versions",
            "ie >= 11"
          ]
        }
      }
    ],
    "@babel/preset-react"
  ],
  "plugins": [
    "<BABEL_PRESET_ROOT>/src/dev-expression.js",
    "<BABEL_PRESET_ROOT>/src/kiwicom-js-invariant.js",
    "<BABEL_PRESET_ROOT>/src/kiwicom-js-warning.js",
    "@babel/plugin-transform-flow-strip-types",
    "@babel/plugin-proposal-class-properties",
    "@babel/plugin-proposal-nullish-coalescing-operator",
    "@babel/plugin-proposal-object-rest-spread",
    "@babel/plugin-proposal-optional-chaining",
    "@babel/plugin-proposal-numeric-separator",
    [
      "@babel/plugin-transform-runtime",
      {
        "useESModules": true
      }
    ],
    "@kiwicom/babel-plugin-orbit-components"
  ],
  "parserOpts": {
    "plugins": [
      "jsx",
      "flow",
      "flowComments",
      "dynamicImport"
    ]
  },
  "retainLines": false
}
`;
